gzip_logs_tools/log_compressor/log_compressor.c

```diff
@@ -138,7 +139,7 @@ file_writer_thread(void* context)
 		}
 				
 		bytes_written = write(output_fd, input_buffer.ptr, input_buffer.size);
-		if (bytes_written != input_buffer.size)
+		if (bytes_written < 0 || (size_t) bytes_written != input_buffer.size) // -1 if not written else size which is <= input_buffer)
```

Comment:
i'd just cast input_buffer.size to ssize_t and avoid the extra check for < 0

Reply:
At first I did that (in ipt.h) but I since I didn't want to make sure this is correct to all the places that it is used I thought it would be simpler/safer to do the check once here

Reply:
how does a function that lets you write size_t bytes returns an ssize_t as result of number of bytes written - oh well

Reply:
this one is still open - just do `if (bytes_written != (ssize_t) input_buffer.size)`